Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+                ANIMATION_FIRST_FRAME, ANIMATION_LAST_FRAME);
+        }
+        for (var i in OVERRIDDEN_DRIVE_KEYS) {
+            MyAvatar.disableDriveKey(OVERRIDDEN_DRIVE_KEYS[i]);
+        }
+
+        MyAvatar.centerBody();
+        var index = MyAvatar.getJointIndex("Hips");
+
+        // var hipLocalRotation = MyAvatar.getJointRotation(index);
+        // var hipWorldRotation = Quat.multiply(MyAvatar.orientation, hipLocalRotation);
+        // var yawHipWorldRotation = Quat.cancelOutRollAndPitch(hipWorldRotation);
+
+        Script.setTimeout(function () {
+
+            print(6, JSON.stringify(chairProperties.rotation));
+
+            if ((HMD.active || testing) && showsPresitOverlay) {
+                Script.update.disconnect(overlays.preSit.update);
+                overlays.preSit.remove();
+            }
+
+            sittingDown = true;
+
+            MyAvatar.pinJoint(index, seatCenterPosition, chairProperties.rotation);
+            canStand = true;
+
+            Script.update.connect(update);
+            MyAvatar.scaleChanged.connect(standUp);
+            location.hostChanged.connect(standUp);
+
+        }, SIT_DELAY);
+
+        animStateHandlerID = MyAvatar.addAnimationStateHandler(function (avatarProperties) {
+            return { headType: 0 };
+        }, ["headType"]);
+
+    }
+
+    function sitDown() {
+        print("SIT DOWN");
+
+        // if (utils.checkSeatForAvatar()) {
+        //     print("Someone is already sitting in that chair.");
+        //     return;
+        // }
+
+        if (overlaySittable !== null) {
+            overlays.sittable.remove();
+        }
+
+        utils.setHeadToHipsDistance();
+
+        lockChairOnStandUp = Entities.getEntityProperties(entityID, 'locked').locked;
+        Entities.editEntity(entityID, { locked: true });
+
+        utils.calculatePinHipPosition();
+
+        Script.setTimeout(function () {
+
+            if ((HMD.active || testing) && showsPresitOverlay) {
+
+                overlays.preSit.create();
+                Script.update.connect(overlays.preSit.update);
+
+                Script.setTimeout(function () {
+                    sitAndPinAvatar();
+                }, OVERLAY_PRESIT_FRAME_DURATION * overlayPreSitLoaded.length);
+
+            } else {
+                sitAndPinAvatar();
+            }
+
+        }, SIT_DELAY);
+
+    };
+
+    function standUp() {
+
+        var avatarDistance = Vec3.distance(MyAvatar.position, seatCenterPosition);
+
+
+        Script.setTimeout(function () {
+            // Move avatar in front of the chair to avoid getting stuck in collision hulls
+            if (avatarDistance < STANDUP_DISTANCE) {
+                var index = MyAvatar.getJointIndex("Hips");
+                MyAvatar.clearPinOnJoint(index);
+                var offset = { x: 0, y: 1.0, z: -1 / 2 - chairProperties.dimensions.z * chairProperties.registrationPoint.z };
```

Comment:
the -1/2 should be replaced by a constant. like CHAIR_DISMOUNT_OFFSET or something.